c++/src/kj/compat/http.c++

```diff
@@ -5716,14 +5716,111 @@ public:
     if (!connectSettings.useTls) {
       KJ_IF_MAYBE(wrapper, settings.tlsContext) {
         KJ_IF_MAYBE(tlsStarter, connectSettings.tlsStarter) {
-          auto refConnection = kj::refcountedWrapper(kj::mv(connection));
-          connection = refConnection->addWrappedRef();
-          kj::Own<kj::AsyncIoStream> ref1 = refConnection->addWrappedRef();
-          Function<kj::Own<kj::AsyncIoStream>(kj::StringPtr)> cb =
+          class TransitionaryAsyncIoStream final: public kj::AsyncIoStream {
```

Comment:
Nitpick: This class feels a bit big to be declared nested inside a function like this.